tmk_core/protocol/chibios/usb_main.c

```diff
@@ -930,9 +930,20 @@ void send_consumer(uint16_t data) {
 #ifdef CONSOLE_ENABLE
 
 int8_t sendchar(uint8_t c) {
-    // The previous implmentation had timeouts, but I think it's better to just slow down
-    // and make sure that everything is transferred, rather than dropping stuff
-    return chnWrite(&drivers.console_driver.driver, &c, 1);
+    static bool timeouted = false;
```

Comment:
Can you add a comment to explain why you distinguish the timeout/non-timeout case please? I.e., why not just always use a timeout?

Reply:
I pushed a commit with a comment explaining how the timeouted logic works.

Reply:
Thank you! Based on your comment, I think the logic might be easier to follow if we rename `timeouted` (describing the mechanism) to `hid_listen_connected` (describing the purpose). What do you think?

Reply:
`timeouted` is only roughly the same as `hid_listen**_not_**connected`. There are cases where it's not exactly that, such as during startup, and in case of a horribly overloaded machine. In my opinion we should not rename it.

Reply:
There was no further request to still rename this after my counterargument, so I'm setting this conversation as resolved.

Reply:
I still think it would be good if we could find a clearer name than `timeouted`, but I only speak as a contributor to the project, so not with any authority :)

Reply:
Invert the boolean logic, and call it `listener_present`?

Reply:
I like that suggestion. I agree `listener_present` is clearer.

Reply:
`timed_out`?

Reply:
I disagree with listener_present being a good name, because the variable is only an **approximation** of whether listener is present:

I logged what actually happens for example during keyboard startup.
First the keyboard accepts up to 128 characters without timing out, independent on whether hid_listen is open or not, because initially the queues are empty. Then it starts dropping characters. Then during usb enumeration, somehow these queues get cleared, or perhaps, the OS driver extracts bytes from the queue. So another 128 bytes get successfully written into the queue without timing out, and then we start timing out again. I don't know where the pending 128 bytes go, but the fact is that writes start not timing out yet again, and if we were to flip the boolean logic, and rename the variable to listener_present, then the name would not accurately represent what is actually happening, cause at this stage hid_listen is still not listening to us, and yet we still behaved as if it was listening already twice.

So right now, during boot, if hid_listen is not started, the following happens:
 - we start in the timed_out = false state.
 - we write the first 128 bytes, that all succeed immediately, because there's space in the queue.
 - the 129th byte has no space, so we wait 5ms, and then we drop the byte
 - a number of following bytes will get dropped immediately, with TIME_IMMEDIATE
 - suddenly there's more space in the queues (for whatever reason, it happens during enumeration)
 - we write 128 more bytes, that all succeed immediately, because there's space in the queue.
 - the next byte we write will time out, so we wait another 5ms, and then we drop the byte.
 - all follow-up writes will be timed out and dropped immediately.

Interestingly, when hid_listen connects, it doesn't print the bytes that are pending in the queue. Doesn't look like hid_listen itself drops any bytes, so I suppose this must be in the OS.

I have renamed `timeouted` to `timed_out` as suggested by @fauxpark . I also made the same rename in the lufa code.
And I also refactored the chibios code I have written to make it shorter, and more readable, and clarified the comments a bit.